frontend/packages/knative-plugin/src/components/overview/RevisionsOverviewList.tsx

```diff
@@ -13,16 +17,41 @@ export type RevisionsOverviewListProps = {
   service: K8sResourceKind;
 };
 
+const MAX_REVS: number = 3;
+
 const RevisionsOverviewList: React.FC<RevisionsOverviewListProps> = ({ revisions, service }) => {
   const canSetTrafficDistribution = useAccessReview({
     group: ServiceModel.apiGroup,
     resource: ServiceModel.plural,
     namespace: service.metadata.namespace,
     verb: 'update',
   });
+
+  const filteredRevisions = (): K8sResourceKind[] => {
+    if (!revisions || !revisions.length) {
+      return [];
+    }
+    const traffic = service?.status?.traffic;
+    const [revWithTraffic, revWithoutTraffic] = _.partition(revisions, (element) => {
+      return _.find(traffic, { revisionName: element.metadata?.name });
+    }).map((element) => _.sortBy(element, ['metadata.creationTimestamp']));
+    return revWithTraffic.length < MAX_REVS
+      ? _.concat(revWithTraffic, revWithoutTraffic.slice(0, MAX_REVS - revWithTraffic.length))
+      : revWithTraffic;
+  };
+
   return (
     <>
       <SidebarSectionHeading text="Revisions" className="revision-overview-list">
+        {revisions?.length > MAX_REVS && (
+          <ResourceLink
+            className="all-revisions-link"
+            kind={referenceForModel(RevisionModel)}
+            displayName={`View all (${revisions.length})`}
+            namespace={revisions[0]?.metadata?.namespace}
```

Comment:
can avoid `[0]` and take namespace from service `service?.metadata?.namespace`

Reply:
done